02-peer-protocol.md

```diff
@@ -492,6 +493,58 @@ to broadcast the commitment transaction to get his funds back and open a new
 channel. To avoid this, the funder should ensure the funding transaction
 confirms in the next 2016 blocks.
 
+## Channel Quiescence
+
+Various fundamental changes, in particular protocol upgrades, are
+easiest on channels where both commitment transactions match, and no
+pending updates are in flight.  We define a protocol to quiesce the
+channel by indicating that "SomeThing Fundamental is Underway".
+
+### `stfu`
+
+1. type: 2 (`stfu`)
+2. data:
+    * [`channel_id`:`channel_id`]
+    * [`u8`:`initiator`]
+
+### Requirements
+
+The sender of `stfu`:
+  - MUST NOT send `stfu` unless `option_quiesce` is negotiated.
+  - MUST NOT send `stfu` if any of the sender's htlc additions, htlc removals
+    or fee updates are pending for either peer.
+  - MUST NOT send `stfu` twice.
+  - if it is replying to an `stfu`:
+    - MUST set `initiator` to 0
+  - otherwise:
+    - MUST set `initiator` to 1
+  - MUST set `channel_id` to the id of the channel to quiesce.
+  - MUST now consider the channel to be quiescing.
+  - MUST NOT send an update message after `stfu`.
+
+The receiver of `stfu`:
+  - if it has sent `stfu` then:
+    - MUST now consider the channel to be quiescent
+  - otherwise:
+    - SHOULD NOT send any more update messages.
+    - MUST reply with `stfu` once it can do so.
+
+Upon disconnection:
+  - the channel is no longer considered quiescent.
+
+### Rationale
+
+The normal use would be to cease sending updates, then wait for all
+the current updates to be acknowledged by both peers, then start
+quiescence.  For some protocols, choosing the initiator matters,
+so this flag is sent.
+
+If both sides send `stfu` simultaneously, they will both set
+`initiator` to `1`, in which case the "initiator" is arbitrarily
+considered to be the channel funder (the sender of `open_channel`).
```

Comment:
Was thinking how to implement this, then realize this gives the funder the advantage to always be the initiator, since he/she can always choose to send `stfu(initiator=1)` after receiving one. Not sure about its implications, think it should be fine if it's always the initiator pays if there's cost involved.

Also think we won't have this issue if `option_simplified_update` is in place.

Reply:
> so there's no benefit to being the initiator

Worth noting though that this construction does require downstream protocols to be aware of this and design it such that it is never advantageous to be the initiator.

Reply:
Definitely! But downstream protocols could also choose to override this behavior and use whatever mechanism they'd like to decide who has which responsibility, for example splicing could decide that it's always the initial channel opener who pays the splice fees, even if they didn't initiate the splice (that would be a bad idea of course, but it's just to showcase that downstream protocols do whatever they'd like, regardless of who is the quiescence "initiator").

Reply:
Think we can still add a note.